src/runtime.rs

```diff
@@ -215,13 +216,13 @@ fn parse_build_args_or_exit() -> BuildArgs {
 
 fn read_buildpack_dir<E: Display + Debug>() -> Result<PathBuf, E> {
     env::var("CNB_BUILDPACK_DIR")
-        .map_err(Error::CannotDetermineBuildpackDirectory)
+        .map_err(|e| LibError::CannotDetermineBuildpackDirectory(e).into())
         .map(PathBuf::from)
 }
 
 fn read_buildpack_toml<BM: DeserializeOwned, E: Display + Debug>() -> Result<BuildpackToml<BM>, E> {
```

Comment:
Is it possible to mix specific errors with generic errors? Could we return a `Result<BuildpackToml<BM>, crate::error::LibError>` here? Would the `into` conversion have to go somewhere else or would it just not be possible?

I'm thinking since we know all the errors that this can produce and none of them come from outside sources this could be more specific, but my understanding about the details of why that might not be possible as well as the tradeoffs are fuzzy.

Reply:
> Is it possible to mix specific errors with generic errors?

Yes and no. In any individual function, you need a single Error type. There a things to help with that though like `map_err` and enums. This is why a lot of errors are enums.

To your point it's probably worth revisiting all these and return `LibError` instead of `crate::error::Error` since it doesn't touch user input.